src/sentry/search/events/fields.py

```diff
@@ -259,6 +260,33 @@ def team_key_transaction_expression(organization_id, team_ids, project_ids):
     ]
 
 
+def calculate_count_if_value(args: Mapping[str, str]) -> Union[float, str]:
+    """Ensures that the type of the third parameter is compatible with the first
+    and cast the value if needed
+    eg. duration = numeric_value, and not duration = string_value
+    """
+    column = args["column"]
+    condition = args["condition"]
+    value = args["value"]
+    if (
+        column == "transaction.duration"
+        or is_duration_measurement(column)
+        or is_span_op_breakdown(column)
+    ):
+        try:
+            value = float(value.strip("'"))
+        except Exception:
+            raise InvalidSearchQuery(f"{value} is not a valid value to compare with {column}")
+
+    # TODO: not supporting field aliases or arrays yet
+    elif column in FIELD_ALIASES or column in ARRAY_FIELDS:
+        raise InvalidSearchQuery(f"{column} not supported by count_if")
+    # At this point only string or tag columns are left
+    elif condition not in ["equals", "notEquals"]:
+        raise InvalidSearchQuery(f"{condition} not compatible with {column}")
```

Comment:
since this column validation has nothing to do with the typing of the value, can this be validated in one of the subclasses of the FunctionArgs class? It we do this kind of validation in the `normalize` method there generally. We  may have to introduce a countIf column type or something though

Reply:
So ended up with this cause the normalize doesn't have the context of the other variables, and I think its better if normalize doesn't need that. So calculated args seemed like a better fit to me